python/tvm/octo/octo_model.py

```diff
@@ -0,0 +1,191 @@
+            tar.add(input_info_path, "input_info.json")
+            tar.add(metadata_path, "metadata.json")
+
+    def load(self, model_path: Union[str, Path]) -> Tuple[relax.vm.Executable, Dict[List, str]]:
+        """Load a saved OctoModel back into memory.
+
+        Parameters
+        ----------
+        model_path : Union[str, Path]
+            The path to the saved OctoModel that will be loaded.
+
+        Returns
+        -------
+        exe : relax.vm.Executable
+            A compiled executable that can be loaded and run by a relax VM.
+        input_info : Dict[str, Tuple[List, str]]
+            Information about the input names, shapes, and types for the VM.
+            Will be loaded from memory if possible.
+        """
+        t = tarfile.open(model_path)
+        t.extractall(self._tmp_dir.relpath("."))
+
+        # Load executable.
+        exe_path = self._tmp_dir.relpath("exe.so")
+        exe = relax.vm.Executable(tvm.runtime.load_module(exe_path))
+
+        # Load input info.
+        input_info_path = self._tmp_dir.relpath("input_info.json")
+        with open(input_info_path, "r") as fi:
+            input_info = json.load(fi)
+
+        # load other metadata.
+        metadata_path = self._tmp_dir.relpath("metadata.json")
+        with open(metadata_path, "r") as fi:
+            metadata = json.load(fi)
+        self.target = tvm.target.Target(metadata["target"])
+
+        return exe, input_info
+
+    def generate_inputs(self) -> Dict[str, np.array]:
+        """Generate random inputs for inference or benchmarking
+
+        Returns
+        -------
+        input_dict : Dict[str, np.array]
+        """
+        input_dict = {}
+        for name, (shape, dtype) in self.input_info.items():
+            input_dict[name] = np.random.normal(size=shape).astype(dtype)
+        return input_dict
+
+    def run(self, inputs: Optional[Dict[str, np.array]] = None) -> List[np.array]:
+        """Perform an inference of the model.
+
+        Parameters
+        ----------
+        inputs : Optional[Dict[str, np.array]]
+            An optional input dictionary containing the values to perform
+            inference with. If not provided, random values will be generated
+            instead.
+
+        Returns
+        -------
+        outputs : List[np.array]
+            The output values from the inference.
+        """
+        # Generate random inputs if none are provided.
+        if inputs is None:
+            inputs = self.generate_inputs()
+
+        # Assign inputs.
+        self.vm.set_input("main", **inputs)
+        # Run the modeel.
+        self.vm.invoke_stateful("main")
+        # Get and return the outputs.
+        outputs = self.vm.get_outputs("main")
+        if isinstance(outputs, tuple):
+            outputs = [output.numpy() for output in outputs]
+        else:
+            outputs = [outputs.numpy()]
+        return outputs
+
+    def profile(self) -> tvm.runtime.profiling.Report:
```

Comment:
I think we can consider this as follow-up. Right now, I dont see it bringing a lot of value since Relax profiling itself doesnt allow much configuration. For example, there isnt an option to change the number of iterations and inputs dont have an effect on the result.